src/classes/ALLO_MakeDefaultAllocations_BATCH.cls

```diff
@@ -134,11 +134,11 @@ public class ALLO_MakeDefaultAllocations_BATCH implements Database.Batchable<sOb
         //No error handling necessary, because the only place this is called from is the widget that displays the batch
         //process status in the UI.
         if (!listAlloForInsert.isEmpty()) {
-            ALLO_Allocations_TDTM.resetFlags(true);
+            ALLO_ProcessControl.setRecursionFlag(true);
```

Comment:
Because these are at the end of batch iteration so there would be no additional actions performed after those dml statements (in the current execution context). Wouldn't harm anything to add them either.